### DISCUSSION TITLE
How to do IPC between two process by sdbus::createServerBus and sdbus::createDirectBusConnection

### DISCUSSION THREAD

**Discussion by kingysu** [Kistler-Group/sdbus-cpp#453] [Q&A]
In the article: [Using direct (peer-to-peer) D-Bus connections](https://github.com/Kistler-Group/sdbus-cpp/blob/master/docs/using-sdbus-c++.md#using-direct-peer-to-peer-d-bus-connections), there is a demo about how to do IPC between two process which are parent and child.

However, I want to do direct D-Bus communication between two different processes which are not parent and child. I have tried to crate a new socket fd with `socket()` function, but when pass the socket fd into `sdbus::createDirectBusConnection`, an error is printed: Invalid argument. Although I know the fd is invalid, but I don't know how to correct it. Is there anyone who can help? Thank you in advance.

**Comment by sangelovic** [comment]
I can't tell you off the top of my head, but I think the socket has to be connected. Have you solved the issue in the meantime?

**Reply by kingysu** [reply]
Sorry, I haven't sovled this problem yet. Because of limited time, we had to use the gdbus API.
